patmat/src/main/scala/patmat/Huffman.scala

```diff
@@ -64,7 +70,10 @@ trait Huffman extends HuffmanInterface:
    *       println("integer is  : "+ theInt)
    *   }
    */
-  def times(chars: List[Char]): List[(Char, Int)] = ???
+  def times(chars: List[Char]): List[(Char, Int)] = chars.groupBy(identity)
+    .view
+    .mapValues(_.size)
+    .toList
```

Comment:
isn't it better to `chars.view[...]`? So `groupBy(identity)` won't eagarly create the list of chars for each group? If you `.view` after `groupBy`, it's just another extra step which doesn't help. Correct me if i'm wrong

Reply:
> isn't it better to `chars.view[...]`? So `groupBy(identity)` won't eagarly create the list of chars for each group? If you `.view` after `groupBy`, it's just another extra step which doesn't help. Correct me if i'm wrong

I think it will create the list anyway. From groupBy documentation:
"Note: Even when applied to a view or a lazy collection it will always force the elements."

Reply:
Wow, this is confusing... it's very strange there's no way of lazily using `groupBy`